gnovm/pkg/gnolang/nocompile_386.go

```diff
@@ -0,0 +1,13 @@
+//go:build 386
```

Comment:
`386` is not the only 32-bit GOARCH, see [this gist](https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63).

Reply:
Morgan's solution is even better, as it will fail at compile time.
See https://github.com/gnolang/gno/pull/3643#discussion_r1945059833